src/main/java/org/folio/tlib/postgres/impl/PgCqlQueryImpl.java

```diff
@@ -0,0 +1,270 @@
+
+  String language = "english";
+  CQLNode cqlNodeRoot;
+
+  @Override
+  public void parse(String query) {
+    if (query == null) {
+      cqlNodeRoot = null;
+    } else {
+      try {
+        log.debug("Parsing {}", query);
+        cqlNodeRoot = parser.parse(query);
+      } catch (CQLParseException | IOException e) {
+        throw new IllegalArgumentException(e.getMessage());
+      }
+    }
+  }
+
+  @Override
+  public String getWhereClause() {
+    return whereHandle(cqlNodeRoot);
+  }
+
+  static String basicOp(CQLTermNode termNode) {
+    String base = termNode.getRelation().getBase();
+    switch (base) {
+      case "==":
+        return "=";
+      case "=":
+      case "<>":
+        return base;
+      default:
+        throw new IllegalArgumentException("Unsupported operator " + base + " for: "
+            + termNode.toCQL());
+    }
+  }
+
+  static String numberOp(CQLTermNode termNode) {
+    String base = termNode.getRelation().getBase();
+    switch (base) {
+      case "==":
+        return "=";
+      case "=":
+      case "<>":
+      case ">":
+      case "<":
+      case "<=":
+      case ">=":
+        return base;
+      default:
+        throw new IllegalArgumentException("Unsupported operator " + base + " for: "
+            + termNode.toCQL());
+    }
+  }
+
+  /**
+   * See if this is a CQL query with a existence check (NULL or NOT NULL).
+   * <p>Empty term makes "IS NULL" for CQL relation =, "IS NOT NULL" for CQL relation <>.
+   * </p>
+   * @param field CQL field.
+   * @param termNode term.
+   * @return SQL op for NULL; null if not a NULL check.
+   */
+  String handleNull(PgCqlField field, CQLTermNode termNode) {
+    if (!termNode.getTerm().isEmpty()) {
+      return null;
+    }
+    String base = termNode.getRelation().getBase();
+    switch (base) {
+      case "==":
+        return null;
+      case "=":
+        return field.getColumn() + " IS NULL";
+      case "<>":
+        return field.getColumn() + " IS NOT NULL";
+      default:
+        throw new IllegalArgumentException("Unsupported operator " + base + " for: "
+            + termNode.toCQL());
+    }
+  }
+
+  String handleTypeUuid(PgCqlField field, CQLTermNode termNode) {
+    String s = handleNull(field, termNode);
+    if (s != null) {
+      return s;
+    }
+    UUID id = UUID.fromString(termNode.getTerm()); // so IllegalArgumentException is thrown
+    String pgTerm = "'" + id + "'";
+    String op = basicOp(termNode);
+    return field.getColumn() + op + pgTerm;
+  }
+
+  String handleTypeText(PgCqlField field, CQLTermNode termNode) {
+    String s = handleNull(field, termNode);
+    if (s != null) {
+      return s;
+    }
+    String base = termNode.getRelation().getBase();
+    boolean ft = "=".equals(base) || "all".equals(base);
+    String cqlTerm = termNode.getTerm();
+    StringBuilder pgTerm = new StringBuilder();
+    boolean backslash = false;
+    for (int i = 0; i < cqlTerm.length(); i++) {
```

Comment:
I would have been inclined to move this backslash-processing out into its own function — not least because it's tricky enough that it would be worth testing in isolation.

Reply:
ok. Done.